### DISCUSSION TITLE
Viewing Image In Full Screen

### DISCUSSION THREAD

**Discussion by gorkijatt** [IacobIonut01/Gallery#136] [Q&A] [answered]
"I need some help. I don't know where to ask questions regarding this. The thing is, I want to view an image in full screen by hiding the navigation and status bars. I used `WindowCompat.setDecorFitsSystemWindows(window, false)`, which started showing content behind the status bar. When I add the flags to hide the status and navigation bars, the content automatically takes margin from the top. I've tried everything available on the internet, but no luck. I can't figure out where I am going wrong. I've been trying this for at least 4 days, and I still haven't had any luck."

**Comment by gorkijatt** [comment]
This is what i tried but the content is still showing 
```
class MainActivity : ComponentActivity() {
    @OptIn(ExperimentalMaterial3WindowSizeClassApi::class)
    override fun onCreate(savedInstanceState: Bundle?) {
        WindowCompat.setDecorFitsSystemWindows(window, false)
        super.onCreate(savedInstanceState)
        setContent {
            MyApplicationTheme {
                val systemUiController = rememberSystemUiController()
                systemUiController.isStatusBarVisible = false // Status bar
                systemUiController.isNavigationBarVisible = false // Navigation bar
                systemUiController.isSystemBarsVisible = false // Status & Navigation bars
                systemUiController.navigationBarDarkContentEnabled = true
                // A surface container using the 'background' color from the theme
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .background(Color.White)
                        .padding(-10.dp),
                ) {
                    Greeting("Android")
                }
            }
        }
    }
}
```

![image](https://github.com/IacobIonut01/Gallery/assets/19510688/8ca6bcbf-ab3d-4179-bf25-b5b743c928f6)
![image](https://github.com/IacobIonut01/Gallery/assets/19510688/fd341142-2bc7-4e07-a721-c02f4079a796)

i want to achieve full screen

**Reply by IacobIonut01** [reply]
Add this flag in your app xml theme
https://github.com/IacobIonut01/Gallery/blob/v1/app/src/main/res/values/themes.xml#L5

**Comment by IacobIonut01** [comment, marked as answer]
So, to control the systemBars I use WindowInsetsController (rememberWindowInsetsController for compose)

https://github.com/IacobIonut01/Gallery/blob/v1/app/src/main/kotlin/com/dot/gallery/feature_node/presentation/util/WindowInsetsControllerExt.kt#L11

Add this to your app xml theme to completely move to Fullscreen 
https://github.com/IacobIonut01/Gallery/blob/v1/app/src/main/res/values/themes.xml#L5

**Reply by gorkijatt** [reply]
Oh man, it works. Setting the cutout in XML is required for notches. Thanks a lot, man.

**Reply by sosauce** [reply]
Wouldn't be much easier to use the edgetoedge API ? just call enableEdgeToEdge() in ur activity 🤔
Maybe im wrong tho ;)

**Reply by gorkijatt** [reply]
yes i am using now edge to edge.. things are getting more easier now with compose
